src/python/pants/backend/python/rules/pex.py

```diff
@@ -219,6 +220,21 @@ def __init__(
         self.additional_args = tuple(additional_args)
 
 
+@dataclass(frozen=True)
+class TwoStepPexRequest:
```

Comment:
Maybe `MergedPexRequest`? We already create two (or more) step PEXes by leveraging PEX_PATH to solve the same problem described here. The distinction here is we want the end result merged instead of PEX_PATH'd together since the final product will be end-user visible.

Reply:
Maybe `PexWithRequirementsOptimizationRequest`. That pained me to write out just now..maybe not.

I think I prefer `TwoStepPexRequest` to `MergedPexRequest` because `MergedPexRequest` sounds more generic than this really is. The only pattern this type supports is the two steps of 1) requirements, 2) merge into sources. It doesn't allow arbitrary merging of pexes.

Reply:
Hmm, I see the point re similar functionality due to PEX_PATH.  On balance, "merged" seems a bit generic and I'm not sure it clarifies much. And TwoStep is at least evocative (a la https://en.wikipedia.org/wiki/Country-western_two-step)...